common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -111,5 +114,28 @@ public static AuthType getAuthTypeFromConf(TachyonConf conf) {
         AuthType.NOSASL.getAuthName()));
   }
 
-  // TODO: add methods of getting different Thrift class in follow-up PR.
+  /**
+   * For server side, this method return a TTransportFactory based on the auth type. It is used as
+   * one argument to build a Thrift TServer.
+   * @return a corresponding TTransportFactory
+   * @throws SaslException if building a TransportFactory fails
+   * @throws UnsupportedOperationException if the auth type is not supported or recognized.
+   */
+  public TTransportFactory getServerTransportFactory() throws SaslException {
+    switch (mAuthType) {
+      case NOSASL:
+        return new TFramedTransport.Factory();
+      case SIMPLE:
```

Comment:
do we intend to return `PlainServerTransport` for both `SIMPLE` and `CUSTOM`? In this case, please comment `// indent to fall through` after `case SIMPLE:`